src/Symfony/Component/HttpClient/Tests/AmpHttpClientTest.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Tests;
+
+use Symfony\Component\HttpClient\AmpHttpClient;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Process;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/*
+Tests for HTTP2 Push need a recent version of both PHP and curl. This docker command should run them:
+docker run -it --rm -v $(pwd):/app -v /path/to/vulcain:/usr/local/bin/vulcain -w /app php:7.3-alpine ./phpunit src/Symfony/Component/HttpClient/Tests/AmpHttpClientTest.php --filter testHttp2Push
+The vulcain binary can be found at https://github.com/symfony/binary-utils/releases/download/v0.1/vulcain_0.1.3_Linux_x86_64.tar.gz - see https://github.com/dunglas/vulcain for source
+*/
+
+class AmpHttpClientTest extends HttpClientTestCase
+{
+    private static $vulcainStarted = false;
+
+    protected function getHttpClient(string $testCase): HttpClientInterface
+    {
+        return new AmpHttpClient();
+    }
+
+    public function testAcceptHeader()
+    {
+        // Amp provides no way to opt-out from sending an Accept header
+        $this->markTestSkipped('todo');
+    }
+
+    public function testInlineAuth()
+    {
+        // Doesn't Amp turn user:pwd in URLs into Authorization headers?
+        $this->markTestSkipped('todo');
```

Comment:
No, we don't do that, because it's deprecated and generally discouraged by the RFC. That's also the reason for throwing if present by default. An interceptor can be built pretty easily.

Reply:
OK, thanks for the hint, I've now implemented the logic.